### ISSUE TITLE
Ex 4 Level 6

### ISSUE THREAD

**Issue by wongjingwen** [nus-cs2030/2223-s2#381]
Does anyone know why the private methods in the Maybe class (get(), isPresent() )produces an error when defined in an anonymous inner class along with the equals method?

**Comment by trwstin** [comment]
I think it's because the compiler cannot confirm that T extends R?

**Comment by wongjingwen** [comment]
Is there a way i can circumvent it?

**Comment by fy17ohhh** [comment]
<img width="509" alt="image" src="https://user-images.githubusercontent.com/103088135/224767405-0ffd81d8-38ec-4a7d-bf35-0c41493d30ed.png">
Hi, I also get similar errors. Does anyone know how to use isEmpty() and get() in the anonymous inner class?

**Comment by granolaluvr69** [comment]
You have to use the other public methods that you implemented previously to get the value from the maybe of temporary maybes or other maybes. I found the .orElse() method particularly useful. You probably only need to change the implementation of the .or(), .equals() and .flatMap() methods.

**Comment by wongjingwen** [comment]
The orElse() method only takes in a Maybe<T> , i cant seem to use a Maybe<? extends T> to call this method.

**Comment by jeffkhong11** [comment]
Is it the equals() in the anonymous inner class? If so, I used orElse() throughout the method

**Comment by wongjingwen** [comment]
Yes the equals in the anonymous class? However orElse gives a return type of T but the other object maynot have a return type of T

**Comment by wongjingwen** [comment]
Hi how did you manage to solve it ?
